### PR TITLE
Setup + updates

### PR DESCRIPTION
**Pull request by filiplazovic**

### CODE REVIEW

**Comment by filiplazovic** [comment]
@mfaghihi Make sure you go through these changes. 2 biggest things to take from this:
1. Make sure you remove unused parameters after the implementation is changed
2. Avoid cloning, unless there is no other option

File: src/keypair/vanchor/mod.rs [hunk 1]
```diff
@@ -1,46 +1,35 @@
...
-
-	pub fn private_key(&self) -> Result<B, Error> {
-		Ok(self.private_key.clone())
+		}
 	}
 
 	pub fn public_key(&self, h: &H2::Parameters) -> Result<H2::Output, Error> {
-		let privkey = self.private_key.clone();
-		let bytes = to_bytes![privkey].unwrap();
+		let bytes = to_bytes![&self.private_key].unwrap();
```
Comment (mfaghihi) [inline src/keypair/vanchor/mod.rs:23]: Should we change these .unwrap()s to "?" ?

Comment (filiplazovic) [reply src/keypair/vanchor/mod.rs:23]: That's right, good catch

File: src/setup/vanchor.rs [hunk 2]
```diff
@@ -11,81 +21,82 @@ use ark_std::{
 pub type PoseidonCRH_x5_5<F> = CRH<F, PoseidonRounds_x5_5>;
 pub type PoseidonCRH_x5_5Gadget<F> = CRHGadget<F, PoseidonRounds_x5_5>;
 
-pub fn generate_vanchor_leaf_rng<
+pub struct VAnchorPoverSetup<
```
Comment (drewstone) [inline src/setup/vanchor.rs:24]:
```suggestion
pub struct VAnchorProverSetup<
```

**Pull request merged**